File: _experimental/_c++/numbers/src/implementations/numbers_recursive.cpp
#include "../interfaces/numbers_recursive.h"

int NumbersRecursive::fibonacci(int n)
{
    if (n <= 1)
    {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int NumbersRecursive::factorial(int n)
{
    if (n <= 1)
    {
        return 1;
    }
    return n * factorial(n - 1);
}

int NumbersRecursive::sum_numbers(int n)
{
    if(n <= 0){
        return 0;
    }
    return n + sum_numbers(n - 1);
}

File: _experimental/_kotlin/numbers/src/main/kotlin/numbers_recursive.kt
class NumbersRecursive {
    fun fibonacci(n: Int): Int {
        return if (n <= 1) n else fibonacci(n - 1) + fibonacci(n - 2)
    }

    fun factorial(n: Int): Int {
        return if (n <= 1) 1 else n * factorial(n - 1)
    }

    fun sumNumbers(n: Int): Int {
        return if (n <= 1) n else n + sumNumbers(n - 1)
    }
}